buckets/redis/bucket.go

```diff
@@ -56,32 +54,30 @@ func (a *abstractBucket) Take(ctx context.Context, requested int64, maxWaitTime
 		strconv.FormatInt(requested, 10), strconv.FormatInt(maxWaitTime.Nanoseconds(), 10),
 		maxIdleTimeMillis, a.maxDebtNanos}
 
-	var waitTime time.Duration
-	var err error
-
 	client := a.factory.Client().(*redis.Client)
 	res := a.takeFromRedis(ctx, client, args)
-	switch waitTimeNanos := res.Val().(type) {
+	if err := res.Err(); err != nil {
+		if isRedisClientClosedError(err) {
+			fmt.Println("reconnecting")
```

Comment:
Should this be some kind of log?

Reply:
Whoops, one of my debug messages. I can turn it into a log 👍